moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1311,27 +1300,39 @@ as the new values that correspond to the group */
   /** \brief Update the state after setting a particular link to the input global transform pose.*/
   void updateStateWithLinkAt(const LinkModel* link, const Eigen::Isometry3d& transform, bool backward = false);
 
-  const Eigen::Isometry3d& getGlobalLinkTransform(const std::string& link_name)
-  {
-    return getGlobalLinkTransform(robot_model_->getLinkModel(link_name));
-  }
+  /** \brief Get the link transform w.r.t. the root link (model frame) of the RobotModel.
+   *   This is typically the root link of the URDF unless a virtual joint is present.
+   *   Checks the cache and if there are any dirty (non-updated) transforms, first updates them as needed
+   *   A related, more comprehensive function is |getFrameTransform| which includes this functionality
+   *   but also other frame sources (attached objects, subframes).
+   */
+  const Eigen::Isometry3d& getGlobalLinkTransform(const std::string& link_name);
 
-  const Eigen::Isometry3d& getGlobalLinkTransform(const LinkModel* link)
-  {
-    updateLinkTransforms();
-    return global_link_transforms_[link->getLinkIndex()];
-  }
+  const Eigen::Isometry3d& getGlobalLinkTransform(const LinkModel* link);
 
-  const Eigen::Isometry3d& getCollisionBodyTransform(const std::string& link_name, std::size_t index)
-  {
-    return getCollisionBodyTransform(robot_model_->getLinkModel(link_name), index);
-  }
+  const Eigen::Isometry3d& getGlobalLinkTransform(const std::string& link_name) const;
 
-  const Eigen::Isometry3d& getCollisionBodyTransform(const LinkModel* link, std::size_t index)
-  {
-    updateCollisionBodyTransforms();
-    return global_collision_body_transforms_[link->getFirstCollisionBodyTransformIndex() + index];
-  }
+  const Eigen::Isometry3d& getGlobalLinkTransform(const LinkModel* link) const;
+
+  /** \brief Get the link transform w.r.t. the root link (model frame) of the RobotModel.
+   *   This is typically the root link of the URDF unless a virtual joint is present.
+   *   Checks the cache and if there are any dirty (non-updated) transforms first and updates transforms cache as needed
+   *   A related, more comprehensive function is |getFrameTransform| which includes this functionality
+   *   but also other frame sources (attached objects, subframes).
+   *
+   *   As opposed to the visual links in |getGlobalLinkTransform|, this function returns
+   *   the collision link transform used for collision checking.
```

Comment:
This helps explain when to use which one. I find this distinction still a bit confusing. Is this the difference:

- global transform for link: transform for given link in root coordinate frame of robot model
- collision body transform: transform for given collision body in given link's frame

Your high-level description of the transforms returned by the two families methods is the same. I'd reword them a bit to clarify.

Reply:
They are worded the same because they behave the same. I don't want to spend any more time on this documentation that no one else has bothered to do in the last 8 years. I need to get actual work done.